adapters/xeworks/xeworks.go

```diff
@@ -0,0 +1,156 @@
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := new(adapter)
+	bidder.endpoint = template
+
+	return bidder, nil
+}
+
+func (a *adapter) buildEndpointFromRequest(bidRequest *openrtb2.BidRequest) (string, error) {
+	impExtRaw := bidRequest.Imp[0].Ext
+	var impExt adapters.ExtImpBidder
+
+	if err := json.Unmarshal(impExtRaw, &impExt); err != nil {
+		return "", &errortypes.BadInput{
+			Message: "Bidder impression extension can't be deserialized",
+		}
+	}
+
+	var xeworksExt openrtb_ext.ExtXeworks
+	if err := json.Unmarshal(impExt.Bidder, &xeworksExt); err != nil {
+		return "", &errortypes.BadInput{
+			Message: "Xeworks extenson can't be deserialized",
+		}
+	}
+
+	endpointParams := macros.EndpointTemplateParams{
+		Host:     xeworksExt.Env,
+		SourceId: xeworksExt.Pid,
+	}
+
+	return macros.ResolveMacros(a.endpoint, endpointParams)
+}
+
+func (a *adapter) MakeRequests(
+	openRTBRequest *openrtb2.BidRequest,
+	requestInfo *adapters.ExtraRequestInfo,
+) ([]*adapters.RequestData, []error) {
+	endpoint, err := a.buildEndpointFromRequest(openRTBRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	body, err := json.Marshal(openRTBRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	return []*adapters.RequestData{{
+		Method:  http.MethodPost,
+		Body:    body,
+		Uri:     endpoint,
+		Headers: headers,
+	}}, nil
+}
+
+func (a *adapter) MakeBids(
+	openRTBRequest *openrtb2.BidRequest,
+	requestToBidder *adapters.RequestData,
+	bidderRawResponse *adapters.ResponseData,
+) (*adapters.BidderResponse, []error) {
+	if bidderRawResponse.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if bidderRawResponse.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Bad Request. %s", string(bidderRawResponse.Body)),
+		}}
+	}
+
+	if bidderRawResponse.StatusCode == http.StatusServiceUnavailable {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Bidder unavailable. Please contact the bidder support.",
+		}}
+	}
+
+	if bidderRawResponse.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Status Code: [ %d ] %s", bidderRawResponse.StatusCode, string(bidderRawResponse.Body)),
+		}}
+	}
+
+	responseBody := bidderRawResponse.Body
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(responseBody, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	if len(bidResp.SeatBid) == 0 {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: "Array SeatBid cannot be empty",
+		}}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
+
+	bids := bidResp.SeatBid[0].Bid
+
+	if len(bids) == 0 {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: "Array SeatBid[0].Bid cannot be empty",
+		}}
+	}
```

Comment:
Your bidder can return multiple seatBids. I would recommend to please handle this scenario using for loop. Using for loop will also work for single/multiple seatBids. [Here is an example](https://github.com/prebid/prebid-server/blob/c110c82b9beb292896739335435693fa35a5fa33/adapters/appnexus/appnexus.go#L180)